src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -62,7 +62,22 @@ export class UiSettingsClient implements IUiSettingsClient {
       this.cache['theme:enableUserControl']?.userValue ??
       this.cache['theme:enableUserControl']?.value
     ) {
-      this.cache = defaultsDeep(this.cache, this.getBrowserStoredSettings());
+      const allKeys = new Set([
+        ...Object.keys(this.getBrowserStoredSettings()),
+        ...Object.keys(this.cache),
+      ]);
+      const merged: Record<string, PublicUiSettingsParams & UserProvidedValues> = {};
+      for (const key of allKeys) {
+        const preferBrowser = this.cache[key]?.preferBrowserSetting ?? false;
+        if (preferBrowser) {
+          // browser level setting should have a higher priority, otherwise its userValue could not be consumed.
+          // Only use this pattern for declared setting for security reasons.
+          merged[key] = defaultsDeep({}, this.getBrowserStoredSettings()[key], this.cache[key]);
```

Comment:
Correct me if I'm wrong, if browser level setting should have a higher priority, should browser setting override `this.cache[key]`?
```suggestion
          merged[key] = defaultsDeep({}, this.cache[key], this.getBrowserStoredSettings()[key]);
```

Reply:
In `defaultsDeep`, value on the left has a higher priority.

`
const result = defaultsDeep({}, {"theme:darkMode":{userValue:true}},  {"theme:darkMode":{userValue:false}});
`